communication/mavlink_waypoint_handler.cpp

```diff
@@ -186,20 +164,20 @@ void Mavlink_waypoint_handler::receive_count(Mavlink_waypoint_handler* waypoint_
         if (waypoint_handler->waypoint_receiving_ == false)
         {
             // comment these lines if you want to add new waypoints to the list instead of overwriting them
-            waypoint_handler->waypoint_onboard_count_ = 0;
             waypoint_handler->waypoint_count_ = 0;
             //---//
 
             if ((packet.count + waypoint_handler->waypoint_count_) > MAX_WAYPOINTS)
             {
                 packet.count = MAX_WAYPOINTS - waypoint_handler->waypoint_count_;
             }
-            waypoint_handler->waypoint_count_ =  packet.count + waypoint_handler->waypoint_count_;
+            waypoint_handler->requested_waypoint_count_ = packet.count;
+
             print_util_dbg_print("Receiving ");
             print_util_dbg_print_num(packet.count, 10);
             print_util_dbg_print(" new waypoints. ");
             print_util_dbg_print("New total number of waypoints:");
-            print_util_dbg_print_num(waypoint_handler->waypoint_count_, 10);
+            print_util_dbg_print_num(packet.count + waypoint_handler->waypoint_count_, 10);
             print_util_dbg_print("\r\n");
 
             waypoint_handler->waypoint_receiving_   = true;
```

Comment:
done in https://github.com/lis-epfl/MAVRIC_Library/pull/331/commits/93b59181b3f75d37c40d2a1766df84410101e236